spec/lib/brexit_checker/question_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe BrexitChecker::Question do
+  let(:q1) { FactoryBot.build(:brexit_checker_question) }
+  let(:q2) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(a b) }]) }
+  let(:q3) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(c d) }]) }
+  let(:q4) { FactoryBot.build(:brexit_checker_question, criteria: %w(c)) }
+
+  let(:questions) { [q1, q2, q3, q4] }
+
+  describe '#resolve' do
+    before do
+      allow(BrexitChecker::Question).to receive(:load_all) { questions }
+    end
+
+    let(:subject) {
+      BrexitChecker::Question.resolve(
+        previous_question: previous_question, criteria_keys: criteria_keys
+      )
+    }
+
+    let(:criteria_keys) { [] }
+    let(:previous_question) { nil }
+
+    context 'the page on the form is 0' do
+      let(:previous_question) { 0 }
+      it 'returns question one ' do
+        expect(subject).to eq(q1)
+      end
+    end
+
+    context 'the page on the form is 1 and the criteria_keys match question_two' do
+      let(:previous_question) { 1 }
+      let(:criteria_keys) { %w[c d] }
```

Comment:
I wonder if we should stub the `show?` method of each question, so that the test isn't so tied to the Criteria::Evaluator? At first I didn't notice it, but when I got down here I was wondering why these particular keys would give question 3, and had to scroll up to the top and parse the expression. Some of that seems necessary, so maybe it's just a matter of moving the `let`'s a bit closer to the test.

Reply:
I've stubbed `show?`. Also refactored the examples so hopefully they make a bit more sense now.